src/main/java/gregtech/api/capability/impl/BoilerRecipeLogic.java

```diff
@@ -0,0 +1,282 @@
+    }
+
+    @Override
+    public MetaTileEntityLargeBoiler getMetaTileEntity() {
+        return (MetaTileEntityLargeBoiler) super.getMetaTileEntity();
+    }
+
+    @Override
+    protected void setActive(boolean active) {
+        if (active != this.isActive) {
+            getMetaTileEntity().replaceFireboxAsActive(active);
+        }
+        super.setActive(active);
+    }
+
+    @Override
+    public NBTTagCompound serializeNBT() {
+        NBTTagCompound compound = super.serializeNBT();
+        compound.setInteger("Heat", currentHeat);
+        compound.setInteger("ExcessFuel", excessFuel);
+        compound.setInteger("ExcessWater", excessWater);
+        compound.setInteger("ExcessProjectedEU", excessProjectedEU);
+        return compound;
+    }
+
+    @Override
+    public void deserializeNBT(NBTTagCompound compound) {
+        super.deserializeNBT(compound);
+        this.currentHeat = compound.getInteger("Heat");
+        this.excessFuel = compound.getInteger("ExcessFuel");
+        this.excessWater = compound.getInteger("ExcessWater");
+        this.excessProjectedEU = compound.getInteger("ExcessProjectedEU");
+    }
+
+    @Override
+    public void writeInitialData(PacketBuffer buf) {
+        super.writeInitialData(buf);
+        buf.writeVarInt(currentHeat);
+        buf.writeVarInt(lastTickSteamOutput);
+    }
+
+    @Override
+    public void receiveInitialData(PacketBuffer buf) {
+        super.receiveInitialData(buf);
+        this.currentHeat = buf.readVarInt();
+        this.lastTickSteamOutput = buf.readVarInt();
+    }
+
+    @Override
+    public void receiveCustomData(int dataId, PacketBuffer buf) {
+        super.receiveCustomData(dataId, buf);
+        if (dataId == BOILER_HEAT) {
+            this.currentHeat = buf.readVarInt();
+        } else if (dataId == BOILER_LAST_TICK_STEAM) {
+            this.lastTickSteamOutput = buf.readVarInt();
+        }
+    }
+
+    // Required overrides to use RecipeLogic, but all of them are redirected by the above overrides.
+
+    @Override
+    protected long getEnergyInputPerSecond() {
+        GTLog.logger.error("Large Boiler called getEnergyInputPerSecond(), this should not be possible!");
+        return 0;
+    }
+
+    @Override
+    protected long getEnergyStored() {
+        GTLog.logger.error("Large Boiler called getEnergyStored(), this should not be possible!");
+        return 0;
+    }
+
+    @Override
+    protected long getEnergyCapacity() {
+        GTLog.logger.error("Large Boiler called getEnergyCapacity(), this should not be possible!");
+        return 0;
+    }
+
+    @Override
+    protected boolean drawEnergy(int recipeEUt) {
+        GTLog.logger.error("Large Boiler called drawEnergy(), this should not be possible!");
+        return false;
+    }
+
+    @Override
+    protected long getMaxVoltage() {
+        GTLog.logger.error("Large Boiler called getMaxVoltage(), this should not be possible!");
+        return 0;
```

Comment:
This could return the maximum EU generated per boiler type. I think Tech's generator PR uses this for display, or maybe I am thinking incorrectly.

Reply:
I'm sure we could do this, but I would rather keep the method names more representative of what they are